conu/backend/docker/backend.py

```diff
@@ -47,3 +47,8 @@ def cleanup_containers():
             logger.debug("Removing container %s created by conu", id)
             client.stop(id)
             client.remove_container(id)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        super(DockerBackend, self).__exit__(exc_type, exc_val, exc_tb)
+        if self.cleanup:
+            DockerBackend.cleanup_containers()
```

Comment:
It's probably out of scope of this PR but it would be awesome if this code would live in Backend class and was usable for every backend.